plugins/inputs/kinesis_consumer/kinesis_consumer_test.go

```diff
@@ -0,0 +1,178 @@
+			fields: fields{
+				ContentEncoding: "",
+				parser:          parser,
+				records:         make(map[telegraf.TrackingID]string),
+			},
+			args: args{
+				r: &consumer.Record{Data: notZippedBytes, SequenceNumber: aws.String("anything")},
+			},
+			wantErr: false,
+			expected: expected{
+				messageContains: "bob",
+				numberOfMetrics: 2,
+			},
+		},
+		{
+			name: "test no compression via identity ContentEncoding",
+			fields: fields{
+				ContentEncoding: "identity",
+				parser:          parser,
+				records:         make(map[telegraf.TrackingID]string),
+			},
+			args: args{
+				r: &consumer.Record{Data: notZippedBytes, SequenceNumber: aws.String("anything")},
+			},
+			wantErr: false,
+			expected: expected{
+				messageContains: "bob",
+				numberOfMetrics: 2,
+			},
+		},
+		{
+			name: "test no compression via no ContentEncoding",
+			fields: fields{
+				parser:  parser,
+				records: make(map[telegraf.TrackingID]string),
+			},
+			args: args{
+				r: &consumer.Record{Data: notZippedBytes, SequenceNumber: aws.String("anything")},
+			},
+			wantErr: false,
+			expected: expected{
+				messageContains: "bob",
+				numberOfMetrics: 2,
+			},
+		},
+		{
+			name: "test gzip compression",
+			fields: fields{
+				ContentEncoding: "gzip",
+				parser:          parser,
+				records:         make(map[telegraf.TrackingID]string),
+			},
+			args: args{
+				r: &consumer.Record{Data: gzippedBytes, SequenceNumber: aws.String("anything")},
+			},
+			wantErr: false,
+			expected: expected{
+				messageContains: "bob",
+				numberOfMetrics: 1,
+			},
+		},
+		{
+			name: "test zlib compression",
+			fields: fields{
+				ContentEncoding: "zlib",
+				parser:          parser,
+				records:         make(map[telegraf.TrackingID]string),
+			},
+			args: args{
+				r: &consumer.Record{Data: zlibBytpes, SequenceNumber: aws.String("anything")},
+			},
+			wantErr: false,
+			expected: expected{
+				messageContains: "bob",
+				numberOfMetrics: 1,
+			},
+		},
+	}
+
+	k := &KinesisConsumer{
+		ContentEncoding: "notsupported",
+	}
+	err := k.Init()
+	assert.NotNil(t, err)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			k := &KinesisConsumer{
+				ContentEncoding: tt.fields.ContentEncoding,
+				parser:          tt.fields.parser,
+				records:         tt.fields.records,
+			}
+			err := k.Init()
+			assert.Nil(t, err)
+
+			acc := testutil.Accumulator{}
+			if err := k.onMessage(acc.WithTracking(tt.expected.numberOfMetrics), tt.args.r); (err != nil) != tt.wantErr {
+				t.Errorf("onMessage() error = %v, wantErr %v", err, tt.wantErr)
+			}
+
+			assert.Equal(t, tt.expected.numberOfMetrics, len(acc.Metrics))
+
+			for _, metric := range acc.Metrics {
+				if logEventMessage, ok := metric.Fields["message"]; ok {
+					assert.Contains(t, logEventMessage.(string), tt.expected.messageContains)
+				} else {
+					t.Errorf("Expect logEvents to be present")
+				}
+			}
+		})
+	}
+
```

Comment:
@tootedom I think the linter is complaining about this empty line.  We need CI to pass before we can merge, including the linter.
```suggestion
```

Reply:
Many thanks for the pointer.  I was failing miserably at installing revive locally (still not fathomed it).  
Have removed the line, and pushed.  Linter looks happy.